stix2/constants.py

```diff
@@ -0,0 +1,135 @@
+import base64
+import binascii
+import re
+
+# TODO: REConstant?
+# TODO: Timestamp
+
+
+class Constant(object):
+    def __str__(self):
+        return "%s" % self.value
+
+    @staticmethod
+    def escape_quotes_and_backslashes(s):
+        return s.replace(u'\\', u'\\\\').replace(u"'", u"\\'")
+
+
+class StringConstant(Constant):
+    def __init__(self, value):
+        self.value = value
+
+    def __str__(self):
+        return "'%s'" % StringConstant.escape_quotes_and_backslashes(self.value)
+
+
+class IntegerConstant(Constant):
+    def __init__(self, value):
+        try:
+            self.value = int(value)
+        except Exception:
+            raise ValueError("must be an integer.")
+
+
+class FloatConstant(Constant):
+    def __init__(self, value):
+        try:
+            self.value = float(value)
+        except Exception:
+            raise ValueError("must be an float.")
+
+
+class BooleanConstant(Constant):
+    def __init__(self, value):
+        if isinstance(value, bool):
+            self.value = value
+
+        trues = ['true', 't']
+        falses = ['false', 'f']
+        try:
+            if value.lower() in trues:
+                self.value = True
+            if value.lower() in falses:
+                self.value = False
+        except AttributeError:
+            if value == 1:
+                self.value = True
+            if value == 0:
+                self.value = False
+
+        raise ValueError("must be a boolean value.")
+
+
+_HASH_REGEX = {
+    "MD5": ("^[a-fA-F0-9]{32}$", "MD5"),
+    "MD6": ("^[a-fA-F0-9]{32}|[a-fA-F0-9]{40}|[a-fA-F0-9]{56}|[a-fA-F0-9]{64}|[a-fA-F0-9]{96}|[a-fA-F0-9]{128}$", "MD6"),
+    "RIPEMD160": ("^[a-fA-F0-9]{40}$", "RIPEMD-160"),
+    "SHA1": ("^[a-fA-F0-9]{40}$", "SHA-1"),
+    "SHA224": ("^[a-fA-F0-9]{56}$", "SHA-224"),
+    "SHA256": ("^[a-fA-F0-9]{64}$", "SHA-256"),
+    "SHA384": ("^[a-fA-F0-9]{96}$", "SHA-384"),
+    "SHA512": ("^[a-fA-F0-9]{128}$", "SHA-512"),
+    "SHA3224": ("^[a-fA-F0-9]{56}$", "SHA3-224"),
+    "SHA3256": ("^[a-fA-F0-9]{64}$", "SHA3-256"),
+    "SHA3384": ("^[a-fA-F0-9]{96}$", "SHA3-384"),
+    "SHA3512": ("^[a-fA-F0-9]{128}$", "SHA3-512"),
+    "SSDEEP": ("^[a-zA-Z0-9/+:.]{1,128}$", "ssdeep"),
+    "WHIRLPOOL": ("^[a-fA-F0-9]{128}$", "WHIRLPOOL"),
+}
+
+
+class HashConstant(StringConstant):
```

Comment:
Yes - I was working on something to determine what the constant should be based on the object path, as we had discussed - because I thought it would be relatively easy - but there are some inconsistencies with the Property classes - so I decided to go with stuff like this.